portal/src/pages/my/privacy.vue

```diff
@@ -145,20 +153,50 @@ export default {
       } else return false
     },
     submitButtonLabel() {
-      if (
-        this.$store.getters.hasGivenCommunityPermission !==
-        this.hasInitialCommunityPermission
+      if (this.withdrawnCommunityPermission) {
+        return this.$t('Delete-account-and-login')
+      } else if (
+        this.hasGivenCommunityPermission !== this.hasInitialCommunityPermission
       ) {
         return this.$t('save-privacy-settings-and-logout')
       }
       return this.$t('save-privacy-settings')
+    },
+    withdrawnCommunityPermission() {
+      return (
+        !this.hasGivenCommunityPermission && this.hasInitialCommunityPermission
+      )
     }
   },
   mounted() {
-    this.hasInitialCommunityPermission = this.$store.getters.hasGivenCommunityPermission
+    this.hasInitialCommunityPermission = this.hasGivenCommunityPermission
   },
   methods: {
+    deleteAccount() {
+      this.isSubmitting = true
+      this.$store
+        .dispatch('deleteUser')
+        .then(() => {
+          this.$store.dispatch('logout', { fully: true })
+        })
+        .catch(error => {
+          if (error) {
```

Comment:
Eventueel kun je hier nog checken wat de status code van de error is. De session expired is alleen op een 401. Wanneer je een 500 of 502 krijgt, dan wil je misschien niet de indruk wekken, dat het aan je sessie ligt.

Maar dit is een detail. Het is al veel beter dan helemaal geen bericht. We kunnen dit dus ook mergen wat mij betreft.